### PR TITLE
Add google provider #13481

### PR DESCRIPTION
**Pull request by martinvy**
I refactored tests, especially database tests. The database connection has to be set from command line while executing the database tests:
`py.test --database-tests --db-connection postgres://postgres:postgres@localhost:5432/golddigger-test`.
I added one test for new provider.

@business-factory/pythonistas please review & merge.

### CODE REVIEW

File: gold_digger/data_providers/google.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from ._provider import Provider
+
+
+class Google(Provider):
+    """
+    Offers only latest exchange rates for only one currency pair at the moment.
+    """
+    BASE_URL = 'https://www.google.com/finance/converter?a=1&from={}&to={}'
+    BASE_CURRENCY = "USD"
+    RESULT_REGEX = re.compile("class=bld>([\d.]+)")
```
Comment (harpener) [inline gold_digger/data_providers/google.py]: @martinvy Not 100% foolproof but it will do I guess :-)

File: requirements.txt [hunk 2]
```diff
@@ -1,12 +1,13 @@
-backports-abc==0.4
+cached-property==1.3.0
 click==6.2
+backports-abc==0.4
```
Comment (harpener) [inline requirements.txt]: @martinvy Why was this moved :D

File: tests/database/__init__.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+
+database_test = pytest.mark.skipif(not pytest.config.getoption("--database-tests"), reason="need --database option to run")
```
Comment (harpener) [inline tests/database/__init__.py]: @martinvy I think you meant `reason="need --database-tests option to run"` ;-)

File: tests/database/conftest.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import pytest
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, scoped_session
+from sqlalchemy_utils import create_database, database_exists, drop_database
+
+from gold_digger.database.db_model import Base
+
+
+@pytest.yield_fixture(scope="module")
```
Comment (harpener) [inline tests/database/conftest.py]: @martinvy From py.test-3.0, the `@pytest.yield_fixture` is deprecated and `@pytest.fixture` already support yield statements :-)

File: tests/test_exchange_rate_manager.py [hunk 5]
```diff
@@ -1,26 +1,16 @@
...
-
-@pytest.fixture
-def logger():
-    return logging.getLogger("gold-digger.tests")
-
-
-@pytest.fixture
-def currencies():
-    return DEFAULT_CONFIG_PARAMS["supported_currencies"]
+from gold_digger.data_providers import CurrencyLayer, GrandTrunk
```
Comment (harpener) [inline tests/test_exchange_rate_manager.py]: @martinvy This is out of order :-)

File: tests/database/conftest.py (hunk 4 above)
Comment (martinvy) [inline tests/database/conftest.py]: Thanks, I didn't know that. Actually I took this code from `hildy`. I will take a look.

File: gold_digger/data_providers/google.py (hunk 1 above)
Comment (martinvy) [inline gold_digger/data_providers/google.py]: @harpener I know you wrote you don't like it in the previous PR, but I think it is not necessary to match so many characters before the rate if those few characters are unique.
`<div id=currency_converter_result>1 EUR = <span class=bld>26.0434 CZK`
